pkgs/servers/typedb/default.nix

```
@@ -0,0 +1,99 @@
+{ stdenv, lib, openjdk,typedbHome ? "~/.typedb_home", fetchzip, fetchurl}:
+
+let 
+  typedbVersion = "2.1.1";
+  typedbDirLinux = "typedb-all-linux-${typedbVersion}";
+  typedbDirMac = "typedb-all-mac-${typedbVersion}";
+  typedbDirWindows = "typedb-all-windows-${typedbVersion}";
+  typedbDir = if stdenv.hostPlatform.isWindows then typedbDirWindows
+              else if stdenv.isDarwin          then typedbDirMac
+                                               else typedbDirLinux;
```

```suggestion
  typedbDirs.x86_64-linux = "typedb-all-linux-${typedbVersion}";
  typedbDirs.x86_64-darwin = "typedb-all-mac-${typedbVersion}";
  typedbDirs.x86_64-windows = "typedb-all-windows-${typedbVersion}";
  typedbDir = typedbDirs.${system} or throw "unsupported system ${system}";
```

and same for the url

---

I pulled the url and dir stuff together as you proposed

---

However, line 8 throws an error:
```
error: attempt to call something which is not a function but a set, at /home/faebl/programming/grakn_haskell/lastStart/nixpkgs/pkgs/servers/typedb/default.nix:28:19
(use '--show-trace' to show detailed location information)
```
why can't I do this 'dynamic selection' from an attribute set here? (see the new code in the commit)

---

Explained in new comment